test/powershell/Modules/Microsoft.PowerShell.Management/Resolve-Path.Tests.ps1

```diff
@@ -23,4 +23,30 @@ Describe "Resolve-Path returns proper path" -Tag "CI" {
         $result = Resolve-Path -LiteralPath "TestDrive:\\\\\"
         ($result.Path.TrimEnd('/\')) | Should Be "TestDrive:"
     }
+    It "Resolve-Path -Relative should return correct path on different drive" {
+        $base = Join-Path "TestDrive:" "ResolvePath.relative"
+        $root = Join-Path $base "fakeroot"
+        $file = Join-Path $root "file.txt"
+        $driveName = "RvpaTest"
+        $null = New-Item -Path $base -ItemType Directory -Force
+        $null = New-Item -Path $root -ItemType Directory -Force
+        $null = New-Item -Path $file -ItemType File -Force
+        $null = New-PSDrive -Name $driveName -PSProvider FileSystem -Root $root
+        $driveRoot = Join-Path "$driveName`:" ""
+        $driveFile = Join-Path "$driveName`:" "file.txt"
+        try {
+            Push-Location -Path $driveRoot
+            Resolve-Path -Path $base -Relative | Should Be $base
+        }
+        finally {
+            Pop-Location
+        }
+        try {
+            Push-Location -Path $base
+            Resolve-Path -Path $driveFile -Relative | Should Be $(Resolve-Path -Path $file -Relative)
```

Comment:
I would prefer to not use `Resolve-Path` in the `Should Be` part. Instead have literal string.